src/ethereum/EthereumTools.ts

```diff
@@ -381,6 +382,69 @@ export class EthereumTools implements EdgeCurrencyTools {
     return Object.keys(ethereumPlugins).map(plugin => `wallet:${plugin}`)
   }
 
+  async getTokenDetails(filter: EdgeGetTokenDetails): Promise<EdgeToken[]> {
+    const { contractAddress } = filter
+    if (contractAddress == null) return []
+
+    const valid = EthereumUtil.isValidAddress(contractAddress)
```

Comment:
Question: do we need to do any `0x` stuff to this?

Reply:
no, isValidAddress enforces the 0x requirement